types/time.go

```diff
@@ -2225,9 +2203,21 @@ func GetFormatType(format string) (isDuration, isDate bool) {
 			isDuration, isDate = false, false
 			break
 		}
-		if _, ok := durationTokens[token]; ok {
+		var durationTokens bool
+		var dateTokens bool
+		if len(token) >= 2 && token[0] == '%' {
```

Comment:
You can merge the two switch case block.

Reply:
done.